core/src/test/scala/unit/kafka/integration/BaseTopicMetadataTest.scala

```diff
@@ -201,7 +201,7 @@ abstract class BaseTopicMetadataTest extends ZooKeeperTestHarness {
                                 2000, 0)
         metadata.topicsMetadata.nonEmpty &&
           metadata.topicsMetadata.head.partitionsMetadata.nonEmpty &&
-          expectedIsr == metadata.topicsMetadata.head.partitionsMetadata.head.isr
+          expectedIsr.sorted == metadata.topicsMetadata.head.partitionsMetadata.head.isr.sorted
```

Comment:
Why not simply use `sortBy(_.id)`? It doesn't seem that we need ordering outside of tests, so I think I'd prefer that we don't introduce the `Ordered` interface in `BrokerEndPoint` until that is the case.